listenbrainz/listenstore/influx_listenstore.py

```diff
@@ -384,20 +386,40 @@ def dump_listens(self, location, dump_time=datetime.today(), threads=None):
                 # get listens from all measurements and write them to files in
                 # a temporary dir before adding them to the archive
                 for user in users:
-                    query = 'SELECT * FROM ' + get_escaped_measurement_name(user['musicbrainz_id'])
+                    username = user['musicbrainz_id']
+                    offset = 0
+                    listens = []
 
-                    # loop until we get this user's listens
+                    # Get this user's listens in chunks
                     while True:
-                        try:
-                            results = self.influx.query(query)
+
+                        # loop until we get this chunk of listens
+                        while True:
+                            try:
+                                result = self.influx.query("""
+                                    SELECT *
+                                      FROM {measurement}
+                                     LIMIT {limit}
+                                    OFFSET {offset}
+                                """.format(
+                                    measurement=get_escaped_measurement_name(username),
+                                    limit=DUMP_CHUNK_SIZE,
+                                    offset=offset,
+                                ))
+                                break
+                            except Exception as e:
+                                self.log.error('Error while getting listens for user %s', user['musicbrainz_id'])
+                                self.log.error(str(e))
+                                time.sleep(3)
+
+                        rows = list(result.get_points(get_measurement_name(username)))
+                        if not rows:
                             break
-                        except Exception as e:
-                            self.log.error('Error while getting listens for user %s', user['musicbrainz_id'])
-                            self.log.error(str(e))
 
-                    listens = []
-                    for row in results.get_points(get_measurement_name(user['musicbrainz_id'])):
-                        listens.append(Listen.from_influx(row).to_api())
+                        for row in rows:
```

Comment:
At this point the rows should be dumped to the file and memory cleared, so the whole listens don't have to be stored in ram.

Reply:
Hmm, on it.